qiskit/algorithms/eigensolvers/vqd.py

```diff
@@ -0,0 +1,431 @@
+
+        self._eval_time = None
+        self._eval_count = 0
+
+    @property
+    def initial_point(self) -> Sequence[float] | None:
+        """Returns initial point."""
+        return self._initial_point
+
+    @initial_point.setter
+    def initial_point(self, initial_point: Sequence[float]):
+        """Sets initial point"""
+        self._initial_point = initial_point
+
+    def _check_operator_ansatz(self, operator: BaseOperator | PauliSumOp):
+        """Check that the number of qubits of operator and ansatz match."""
+        if operator is not None and self.ansatz is not None:
+            if operator.num_qubits != self.ansatz.num_qubits:
+                # try to set the number of qubits on the ansatz, if possible
+                try:
+                    self.ansatz.num_qubits = operator.num_qubits
+                except AttributeError as ex:
+                    raise AlgorithmError(
+                        "The number of qubits of the ansatz does not match the "
+                        "operator, and the ansatz does not allow setting the "
+                        "number of qubits using `num_qubits`."
+                    ) from ex
+
+    @classmethod
+    def supports_aux_operators(cls) -> bool:
+        return True
+
+    def compute_eigenvalues(
+        self,
+        operator: BaseOperator | PauliSumOp,
+        aux_operators: ListOrDict[BaseOperator | PauliSumOp] | None = None,
+    ) -> VQDResult:
+
+        super().compute_eigenvalues(operator, aux_operators)
+
+        # this sets the size of the ansatz, so it must be called before the initial point
+        # validation
+        self._check_operator_ansatz(operator)
+
+        initial_point = validate_initial_point(self.initial_point, self.ansatz)
+
+        bounds = validate_bounds(self.ansatz)
+
+        # We need to handle the array entries being zero or Optional i.e. having value None
+        if aux_operators:
+            zero_op = PauliSumOp.from_list([("I" * self.ansatz.num_qubits, 0)])
+
+            # Convert the None and zero values when aux_operators is a list.
+            # Drop None and convert zero values when aux_operators is a dict.
+            if isinstance(aux_operators, list):
+                key_op_iterator = enumerate(aux_operators)
+                converted = [zero_op] * len(aux_operators)
+            else:
+                key_op_iterator = aux_operators.items()
+                converted = {}
+            for key, op in key_op_iterator:
+                if op is not None:
+                    converted[key] = zero_op if op == 0 else op
+
+            aux_operators = converted
+
+        else:
+            aux_operators = None
+
+        if self.betas is None:
+            if isinstance(operator, PauliSumOp):
+                upper_bound = abs(operator.coeff) * sum(
+                    abs(operation.coeff) for operation in operator
+                )
+                self.betas = [upper_bound * 10] * (self.k)
```

Comment:
I am not sure about this. Having self.betas as None means we will autocompute these based on the operator. If VQD is called again with a different operator they will now not be auto-computed again but set as per the last operator. This looks like its used elsewhere in the energy evaluation. Maybe we can just have a local variable here that is set from self.betas or computed and pass this to _get evaluate energy so its captured in scope for the evaulate_energy function (this would save doing a some private instance var, which would be another solution, but since its only needed in one other place from what I can see the former would avoid a stateful instance var)

Reply:
I made the local variable :)